src/ui/qgsdiagrampropertiesbase.ui

```diff
@@ -1619,6 +1578,51 @@
                     </layout>
                    </widget>
                   </item>
+                  <item row="8" column="0">
+                   <widget class="QgsCollapsibleGroupBox" name="mObstaclesGrpBox">
+                    <property name="title">
+                     <string>Obstacles</string>
+                    </property>
+                    <property name="syncGroup" stdset="0">
+                     <string notr="true">labelplacementgroup</string>
+                    </property>
+                    <layout class="QHBoxLayout" name="horizontalLayout_12">
+                     <property name="leftMargin">
+                      <number>8</number>
+                     </property>
+                     <property name="rightMargin">
+                      <number>8</number>
+                     </property>
+                     <item>
+                      <widget class="QLabel" name="label_8">
+                       <property name="text">
+                         <string>Discourage diagrams and labels from covering features</string>
```

Comment:
@nyalldawson why are labels mentioned here? Do they not have their settings in the labels --> Placement tab already?

Reply:
There's no distinction between the two when placing labels -- so if you had diagrams switched on for a layer and have this setting enabled for the diagrams, then it will also block labels from covering the features, not just diagrams.

Reply:
And if I don't set this one but set the "discourage.." in the labels tab, would it prevent diagrams from covering the feature? IOW, should we add _diagrams_ to the corresponding label in Labels tab?

Reply:
Yes, that's correct